test/steward_loyalty_mint.js

```diff
@@ -0,0 +1,187 @@
+    );
+    await steward.setMintManager(mintManager.address);
+    await steward.listNewTokens(
+      [testToken1.id, testToken2.id],
+      [accounts[0], accounts[0]],
+      [patronageNumerator, patronageNumerator],
+      [testToken1.tokenGenerationRate, testToken2.tokenGenerationRate]
+    );
+  });
+
+  it("steward: loyalty-mint. Checking correct number of tokens are received after holding a token for  100min", async () => {
+    await waitTillBeginningOfSecond();
+    testTokenId1 = testToken1.id;
+    const timeHeld = 100; // In minutes
+    // Person buys a token
+    await steward.buy(testTokenId1, web3.utils.toWei("1", "ether"), {
+      from: accounts[2],
+      value: web3.utils.toWei("1", "ether")
+    });
+
+    // TIME INCREASES HERE BY timeHeld
+    await time.increase(time.duration.minutes(timeHeld));
+    // First token bought from patron [Collect patronage will therefore be called]
+    await steward.buy(testTokenId1, ether("1"), {
+      from: accounts[3],
+      value: ether("2")
+    });
+
+    const expectedTokens = multiTokenCalculator(
+      new BN(timeHeld).mul(new BN(60)).toString(),
+      [
+        {
+          tokenGenerationRate: testToken1.tokenGenerationRate.toString()
+        }
+      ]
+    );
+    const amountOfToken = await erc20.balanceOf(accounts[2]);
+
+    assert.equal(amountOfToken.toString(), expectedTokens.toString());
+  });
+
+  it("steward: loyalty-mint. Checking correct number of tokens received after holding 2 different token for  100min", async () => {
+    await waitTillBeginningOfSecond();
+    testTokenId1 = testToken1.id;
+    testTokenId2 = testToken2.id;
+    const timeHeld = 100; // In minutes
+    // Person buys a token
+    await steward.buy(testTokenId1, web3.utils.toWei("1", "ether"), {
+      from: accounts[2],
+      value: web3.utils.toWei("1", "ether")
+    });
+
+    await steward.buy(testTokenId2, web3.utils.toWei("1", "ether"), {
+      from: accounts[2],
+      value: web3.utils.toWei("1", "ether")
+    });
+
+    // TIME INCREASES HERE BY timeHeld
+    await time.increase(time.duration.minutes(timeHeld));
+    // First token bought from patron [Collect patronage will therefore be called]
+    await steward.buy(testTokenId1, ether("1"), {
+      from: accounts[3],
+      value: ether("2")
+    });
+    await steward.buy(testTokenId2, ether("1"), {
+      from: accounts[3],
+      value: ether("2")
+    });
+
+    const expectedTokens = multiTokenCalculator(
+      new BN(timeHeld).mul(new BN(60)).toString(),
+      [
+        {
+          tokenGenerationRate: testToken1.tokenGenerationRate.toString()
+        },
+        {
+          tokenGenerationRate: testToken2.tokenGenerationRate.toString()
+        }
+      ]
+    );
+    const amountOfToken = await erc20.balanceOf(accounts[2]);
+
+    assert.equal(amountOfToken.toString(), expectedTokens.toString());
+  });
+
+  it("steward: loyalty-mint. Checking correct number of tokens after foreclosure.", async () => {
```

Comment:
it("steward: loyalty-mint. Checking correct number of tokens are recieved/minted after foreclosure.", async () => {